api_core/google/api_core/datetime_helpers.py

```diff
@@ -131,11 +131,11 @@ def from_rfc3339(value):
         in UTC.
 
     """
-    with_nanos = _RFC3339_NANOS.match(value)
-
-    if with_nanos is None:
+    try:
         return datetime.datetime.strptime(value, _RFC3339_MICROS).replace(tzinfo=pytz.utc)
-    else:
+    except ValueError:
+        with_nanos = _RFC3339_NANOS.match(value)
+
```

Comment:
Please restore the previous version:  testing the regex first is faster / clearer than raising the exception.

Reply:
I restored the previous version with `if-else` expression, and ran couple of tests this morning. But as I can see this was the reason for the coverage mistakes earlier. I mean, we don't need to check it that way because `_RFC3339_NANOS` pattern already includes the structure which is needed. I tried to think of alternatives but  it looks the only option for me. If I'm losing something will be thankful for the advice.

Reply:
@Emar-Kar are you saying that putting the if/else you are unable to get 100% coverage?

Reply:
I checked both realizations with `try-except` and `if-else` statements. The first one was rejected by Tres and the second failed [coverage](https://source.cloud.google.com/results/invocations/e3cf662f-d7b2-452a-a450-89f294c8d3e2/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-python%2Fpresubmit%2Fapi_core/log). I guess this situation appears due to the `_RFC3339_NANOS` pattern. It could be applied to the time samples even without nanos and it makes no sense to check it again through `if-else`. On that note, I decided to remove the comparison and let the script just to separate parts as-is.

![image](https://user-images.githubusercontent.com/46078689/64160526-c6357180-ce44-11e9-9b9e-c92c42f44f29.png)

Reply:
It seems that as of now the only change was to remove a check to be certain we matched something with the regex? Removing this could result in None.group, which will fail at runtime.

Reply:
@crwilcox oh, I finally get it, sorry. Then I'll restore `if-else` to raise an exception with `None`. Could you check, please?

![code_sample_3](https://user-images.githubusercontent.com/46078689/64363764-19145200-d01a-11e9-92c3-fb4b0afd63f6.png)